File: peakElement.py
public class PeakElement {

    // Function to find a peak element in an array
    public static int findPeakElement(int[] arr) {
        int n = arr.length;
        int left = 0;
        int right = n - 1;

        while (left <= right) {
            int mid = left + (right - left) / 2;

            // Compare the middle element with its neighbors
            if ((mid == 0 || arr[mid - 1] <= arr[mid]) && (mid == n - 1 || arr[mid + 1] <= arr[mid]))
                return arr[mid];
            else if (mid > 0 && arr[mid - 1] > arr[mid])
                right = mid - 1;
            else
                left = mid + 1;
        }

        return -1; // Indicates peak element not found (Note: You can change the return type as needed)
    }

    public static void main(String[] args) {
        int[] arr = { 1, 3, 20, 4, 1, 0 };
        int peakElement = findPeakElement(arr);
        if (peakElement != -1)
            System.out.println("Peak element: " + peakElement);
        else
            System.out.println("Peak element not found.");
    }
}

File: peakElement.cpp
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
         int n = nums.size();
         int l = 0, r = n-1;

         if(n == 1)return 0;

         if(nums[l] > nums[l+1])return 0;

         if(nums[r] > nums[r-1])return n-1;


         l++, r--;

         while(l <= r){
                
                 int mid = l + (r-l)/2;

                 if(nums[mid] > nums[mid+1] and nums[mid] > nums[mid-1]) return mid;

                 else if(nums[mid] < nums[mid+1])l = mid+1;

                 else r = mid-1;
         }

         return -1;
    }
};
